plasmapy/formulary/parameters.py

```diff
@@ -161,7 +163,10 @@ def mass_density(
 @check_relativistic
 @validate_quantities(density={"can_be_negative": False})
 def Alfven_speed(
-    B: u.T, density: [u.m ** -3, u.kg / u.m ** 3], ion="p+", z_mean=None
+    B: u.T,
+    density: [u.m ** -3, u.kg / u.m ** 3],
+    ion: Union[str, Particle],
```

Comment:
@arichar6 Sorry for the back-and-forth on this, but, as @namurphy says, we should just annotate with `Particle` since that is needed for `@particle_input` to work correctly.

With that said, `@particle_input` should be updated to work with `Union[str, Particle]` since those are the actual types that are expected.  This drives my OCD nuts since when annotated with just `Particle` my IDE throws warnings all over the place when I try to input with a string.  I do have a scheme that I think will accomplish this, so I'll open a separate issue on it.